test/integration/backward_compatible/serial/CloudDiscoveryTest.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
+
+const { expect } = require('chai');
+const RC = require('../../RC');
+const TestUtil = require('../../../TestUtil');
+const { AssertionError } = require('assert');
+const { HazelcastError } = require('../../../../lib/core/HazelcastError');
+const { HazelcastClient} = require("../../../../lib/HazelcastClient");
+const { Client } = require("../../../../lib/index");
+const {FailoverConfigBuilder} = require("../../../../lib/config/FailoverConfigBuilder");
+describe('Invalid Cluster Discovery Token Test', function () {
+    let cluster;
+    let client;
+    let cfg;
+
+    const testFactory = new TestUtil.TestFactory();
+
+    it('invalid_discovery_token', async function () {
+        cfg = {network: {hazelcastCloud: {discoveryToken: 'invalid_discovery_token'}}};
+
+        try {
+            testFactory.newHazelcastClientForSerialTests(cfg);
+        } catch (err) {
+            console.log(err);
+            expect(err).to.be.instanceof(HazelcastError);
+        }
+    });
+
+    it('invalid_discovery_token_with_anycluster', async function () {
```

Comment:
What do you mean by any cluster? 🤔 There's no need to write such a test. If you are writing a serial test, you should ideally start clients only with `testFactory.newHazelcastClientForSerialTests`. Also, this test can be a parallel test. Can you move it?

Reply:
@fatihozer0 

Forget about all my comments about this test file. It is not easy to test our change via starting a client. So we will use a unit test.

I have written the following test:

```js
'use strict';

const { expect } = require('chai');
const { HazelcastError } = require('../../../../lib');
const { ClientConfigImpl } = require('../../../../lib/config/Config');
const { HazelcastCloudDiscovery } = require('../../../../lib/discovery/HazelcastCloudDiscovery');

describe('Cloud Discovery Token Test', function () {
    it('should throw proper error with an invalid token', async function () {
        const clientConfig = new ClientConfigImpl();
        const cloudDiscovery = new HazelcastCloudDiscovery(
            clientConfig.properties['hazelcast.client.cloud.url'], 2 << 31
        );

        try {
            await cloudDiscovery.callService();
        } catch (err) {
            expect(err).to.be.instanceof(HazelcastError);
            expect(err.message).to.include('discovery token is invalid');
        }
    });
});
```

You can add it under `test/unit` folder maybe with a name like `CloudDiscoveryInvalidTokenTest.js`. This test is passing. We can walk through it via huddle.

Reply:
@fatihozer0 You can delete this file now.

Reply:
Done.